File: c++/11004.cpp
#include <stdio.h>
  
using namespace std;

int main() {
  int roadmap = 1;
  int roads;
  int points;
  int ans;
  double neg,pos;

  int a[3000], b[3000], c[3000];
  int x[100], y[100];

  while(scanf("%d", &roads) == 1) {

    if (roads == 0) {
      return 0;
    }

    
    for(int i = 0; i < roads; i++)
      scanf("%d %d %d", &a[i], &b[i], &c[i]);

    
    scanf("%d", &points);
    for(int i = 0; i < points; i++)
      scanf("%d %d", &x[i], &y[i]);

    ans = 0;
    for(int p = 0; p < points; p++) {
      neg = 0;
      pos = 0;
      for(int i = 0; i < roads; i++) {
        if (a[i]*x[p]+b[i]*y[p]+c[i] > 0) {
          pos++;
        } else {
          neg++;
        }
      }
      ans += pos*(pos-1)/2;
      ans += neg*(neg-1)/2;
    } 

    printf("Roadmap %d:\n", roadmap);
    printf("Negative Builders Ltd. will build %d New Roads.\n", ans);
    roadmap++;
  }

  return 0;
}

File: python/11004.py
import sys, math

def main():
	# sys.stdin = open('11004.txt', 'r')
	inp = sys.stdin.readlines()

	index = 0
	count = 0
	while index < len(inp):
		roads_nr = int(inp[index].rstrip())
		index += 1
		count += 1

		if roads_nr == 0:
			return

		roads = []
		for i in range(roads_nr):
			r = inp[index].rstrip().split()
			index += 1
			roads.insert(0,(int(r[0]),int(r[1]),int(r[2])))

		PP_nr = int(inp[index].rstrip())
		index += 1

		PP = []
		for i in range(PP_nr):
			p = inp[index].rstrip().split()
			index += 1
			PP.insert(0,(int(p[0]),int(p[1])))


		'''
		 k<0 om eq för båda vägarna har samma tecken i punkten
		-> hitta antalet par av vägar som har samma tecken
		-> med hjälp av formeln för aritmetisk summa '''
		ans = 0
		for p in PP:
			pos = 0
			neg = 0
			for i in roads:
				if i[0]*p[0]+i[1]*p[1]+i[2] > 0:
					pos += 1
				else:
					neg += 1
			ans += pos*(pos-1)/2
			ans += neg*(neg-1)/2
						
		print('Roadmap {}:'.format(count))
		print('Negative Builders Ltd. will build {} New Roads.'.format(int(ans)))


main()